backend/src/worker_flow.rs

```diff
@@ -53,7 +56,7 @@ pub struct Iterator {
 #[serde(tag = "type")]
 pub enum FlowStatusModule {
     WaitingForPriorSteps,
-    WaitingForEvent { event: String },
+    WaitingForEvents { count: u16, previous_result: Value },
```

Comment:
why do we need to store previous result ?

Reply:
After the flow suspends, if it resumes from pull() it will call `handle_flow()`/`push_next_job_flow()` using `flow.args` not the last step's result. So this needs to use the previous step's result so that `previous_result` in the input transform evaluation is actually the previous result and not the flow input.

Maybe it could query the completed_job's table instead.

Reply:
Yeah I think completed_job table make more sense, the previous step uuid is stored in the state